src/Compress/CompressionAlgorithmInterface.php

```diff
@@ -31,4 +31,7 @@ public function decompress($value);
      * @return string
      */
     public function toString();
+
+    /** @psalm-return ($option is string ? mixed : array<string, mixed>) */
+    public function getOptions(string|null $option = null): mixed;
```

Comment:
IMO `getOptions` shouldn't be part of the interface here: getting stuff from "ready to use" services is IMO bad design.